etl/parse_email_save_attachment/parse_messages.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+import os
+import pytz
+import boto3
+import email
+import tempfile
+from email import policy
+from operator import itemgetter
+from datetime import datetime
+
+
+def get_most_recent_file(bucket_name, prefix):
+    s3 = boto3.client("s3")
+
+    # List objects within the specified bucket and prefix
+    response = s3.list_objects_v2(Bucket=bucket_name, Prefix=prefix)
+
+    # Check if the bucket is empty or the prefix doesn't match any files
+    if "Contents" not in response:
+        return "No files found in the bucket with the specified prefix."
+
+    # Sort the files by last modified date
+    files = sorted(response["Contents"], key=itemgetter("LastModified"), reverse=True)
+
+    # Get the most recent file
+    most_recent_file = files[0]
+    return most_recent_file["Key"]
+
+
+def get_file_content(bucket_name, file_key):
+    s3 = boto3.client("s3")
+
+    # Get the object
+    obj = s3.get_object(Bucket=bucket_name, Key=file_key)
+
+    # Get the object's content
+    content = obj["Body"].read().decode("utf-8")
+    return content
+
+
+def parse_email_from_s3(email_content):
+    # Decode the string into bytes
+    email_content_bytes = email_content.encode("utf-8")
```

Comment:
It might be a good idea to check the sender's email address to verify it's from maximo so if we got junk emails in there it won't cause any havoc

Reply:
This is a great idea, and we should do this via the `emergency-mgmt-data` email filter that defines this inbox in AWS. As soon as we deliver this to the geo team and they put it to use, we'll be able to do that, if not before. I'll be on the lookout for the sender's address to be specified. :+1: